src/components/auth/LoginWithAuthServerPage.tsx

```diff
@@ -75,63 +103,122 @@ export const LoginWithAuthServerPage = () => {
       throw new Error('No init sign in result found.')
     }
 
-    await AuthServerProvider.finishSignIn(initSignInResultRef.current)
-    await authenticate(ProviderType.AUTH_SERVER)
+    try {
+      await AuthServerProvider.finishSignIn(initSignInResultRef.current)
+      window.location.reload()
+    } catch (e) {
+      // This failure is probably due to the request expiring. Which is already handled by the expiration timer.
+    }
   }, [])
 
   const onBack = useCallback(() => {
     initSignInResultRef.current = null
     setView(View.WELCOME)
   }, [])
 
+  const onChangeAccount = useCallback(() => {
+    AuthServerProvider.deactivate()
+    window.location.reload()
+  }, [])
+
+  if (view === View.LOADING) {
+    return <Container left={<Loader active size="huge" />} />
+  }
+
   if (view === View.WELCOME) {
     return (
-      <div className="LoginWithAuthServerPage">
-        <div className="background"></div>
-        <div className="main">
-          <div className="logo"></div>
-          <div className="title">Discover a virtual social world</div>
-          <div className="subtitle">shaped by its community of creators & explorers.</div>
-          <Button disabled={disabled} className="button" primary onClick={onWelcomeStart}>
-            Start
-          </Button>
-        </div>
-      </div>
+      <Container
+        left={
+          <>
+            <div className="logo"></div>
+            <div className="title">Discover a virtual social world</div>
+            <div className="subtitle">shaped by its community of creators & explorers.</div>
+            <Button disabled={disabled} className="button" primary onClick={onWelcomeStart}>
+              Start
+            </Button>
+          </>
+        }
+      />
+    )
+  }
+
+  if (view === View.WELCOME_CONNECTED && connectedAccountRef.current) {
+    return (
+      <Container
+        left={
+          <>
+            <div className="logo"></div>
+            <div className="title">Welcome{connectedNameRef.current ? ` ${connectedNameRef.current}` : ''}!</div>
+            <div className="subtitle">Ready to explore?</div>
+            <div className="button-container">
+              <Button disabled={disabled} className="button" primary onClick={onWelcomeStart}>
+                Start
+              </Button>
+              <Button disabled={disabled} className="button button-secondary" inverted onClick={onChangeAccount}>
+                Use a different account
+              </Button>
+            </div>
+          </>
+        }
+        right={
+          <WearablePreview
+            lockBeta={true}
+            panning={false}
+            disableBackground={true}
```

Comment:
you could use the short version
```suggestion
            disableBackground
```

Reply:
Did it as well for lockBeta